docs: clarify what `nvim_buf_set_name()` does (#27925)
neovim/neovim: runtime/doc/api.txt

nvim_buf_set_name({buffer}, {name})                      *nvim_buf_set_name()*
    Sets the full file name for a buffer

>     Works similar to |:file_f|. If the buffer did have a name, that name

Reviewer: ```suggestion
    Works similarly to |:file_f|. If the buffer did have a name, that name
```

>     Sets the full file name for a buffer
> 
>     Works similar to |:file_f|. If the buffer did have a name, that name
>     becomes the |alternate-file| name. An unlisted buffer is created to hold
>     the old name.

Reviewer 2: Do not repeat docs.

```suggestion
    Sets the full file name for a buffer, like |:file_f|.
```
